Nav Sidebar: show 10 most recently edited posts/pages in sidebar

p-jackson opened the pull request:
**_Note to anyone reading this while preparing a release of the FSE plugin_**

This change doesn't require any release notes. We're developing a new feature which is currently hidden behind a flag.

#### Changes proposed in this Pull Request

* Show 10 most recently modified posts
* Only shows posts where the status has the `show_in_list` flag
* Highlight the item that's active in the editor
* Shows a label for the post status if the item is anything other than published (e.g. [Draft])
* Hide slug if the post doesn't have one yet (drafts that have never been published don't have slugs)
* Styling changes to make things a little closer to mock up (still not the whole way there yet)

<img width="1041" alt="Screenshot 2020-05-29 at 7 29 51 PM" src="https://user-images.githubusercontent.com/1500769/83233457-d5cec480-a1e2-11ea-88a2-229d66e77dfe.png">

#### Testing instructions

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

* Checkout branch
* Create `apps/full-site-editing/.wp-env.override.json` and fill it with this content to opt-in to the feature
```
    {
      "config": {
      	"WPCOM_BLOCK_EDITOR_SIDEBAR": true
      }
    }
```
* `cd apps/full-site-editing && yarn dev`
* Open another terminal and `cd apps/full-site-editing && wp-env start`
* Go to `http://localhost:4013/wp-admin`
* Open a page or post in the calypso block editor
  * Click (W) to open sidebar
  * Should show up to 10 posts
    * Includes published, draft, scheduled etc.
    * Sorted by most recently modified
    * Shouldn't show posts/pages that are in the trash
  * Items should look something like pbAok1-JQ-p2#comment-1676
* Sandbox your favourite test site and test in Calypso (make sure to define `WPCOM_BLOCK_EDITOR_SIDEBAR` in `0-sandbox.php`). It should work there too.

PR doesn't include click handlers for items

roo2 commented:
hmm I'm missing something to test this, I got a FSE stickered site and applied this patch to my sandbox but the (W) logo was hidden, does the blog need another sticker or something? It seems like the `wpcom-block-editor-nav-sidebar` isn't being loaded at all 🤔

p-jackson commented:
Oh sorry I forgot to copy paste some important instructions from other PRs!

> * Checkout branch
> * Create `apps/full-site-editing/.wp-env.override.json` and fill it with this content to opt-in to the feature
> ```
>     {
>       "config": {
>       	"WPCOM_BLOCK_EDITOR_SIDEBAR": true
>       }
>     }
> ```
> * `cd apps/full-site-editing && yarn dev`
> * Open another terminal and `cd apps/full-site-editing && wp-env start`
> * Go to `http://localhost:4013/wp-admin`

> * Sandbox your favourite test site and test in Calypso (make sure to define `WPCOM_BLOCK_EDITOR_SIDEBAR` in `0-sandbox.php`). It should work there too.

Sorry for the wild goose chase. Your test site doesn't need the fse sticker, you opt-in using this developer only flag.

roo2 commented:
hmm thanks for the extra info, I still can't work out how to see this sidebar though 🤔 

I've defined the `WPCOM_BLOCK_EDITOR_SIDEBAR` flag on my `0-sandbox.php`, I now see log statements in  apps/full-site-editing/full-site-editing-plugin/wpcom-block-editor-nav-sidebar/src/wpcom-block-editor-nav-sidebar.tsx getting hit 👍, before they were skipped.
 
But It looks like the (W) button is still hidden for me. I'm accessing the editor via the calypso "Pages" page

<img width="945" alt="Screen Shot 2020-06-04 at 10 49 01 AM" src="https://user-images.githubusercontent.com/22446385/83702798-02eefd00-a651-11ea-8ae6-d8f883d40c05.png">

andrewserong commented:
@roo2 looks like you've got the FSE sticker set on your test site, which activates the old wpcom-FSE (unrelated to the new full site editing work going on in Gutenberg). The legacy FSE doesn't use the W back button, so if you use a different test site, or switch off the sticker, that might get it working for you!

(As an aside, we have altogether too many different usages of the term FSE, each of which means something different 😅)

roo2 commented:
ahh thanks @andrewserong!

roo2 commented:
Hey after all that I was able to find some feedback 😄 

### Sometimes menu does not open on new page
Sometimes, when on an unsaved new page, attempting to open the nav sidebar with the (W) button takes you away from the page immediately. Other times it works as expected 🤔. It happens on new pages and new pages created by "copy page" about 50% of the time for me
![Jun-04-2020 12-13-51](https://user-images.githubusercontent.com/22446385/83707414-4bacb300-a65d-11ea-8f22-878b7749ca4b.gif)

### White on grey contrast
Here's a small visual issue with the contrast of a selected draft page, you can't really read the word "Draft" on some monitors because of the white on grey

<img width="271" alt="Screen Shot 2020-06-04 at 11 52 02 AM" src="https://user-images.githubusercontent.com/22446385/83706091-f1f6b980-a659-11ea-9a15-7d82ae0be811.png">

### Should new pages have a "New Page" entry?
I noticed that when you're creating a new page, it doesn't have an entry highlighted, I'd expect to see one myself, but that might be a matter of opinion 🤔.
<img width="1173" alt="Screen Shot 2020-06-04 at 12 26 53 PM" src="https://user-images.githubusercontent.com/22446385/83708113-d17d2e00-a65e-11ea-89b1-efba6a178d73.png">


### Otherwise good 👍 
I tested pages with/without slugs and scheduled and draft pages and entering the editor from different places and everything else worked as expected

p-jackson commented:
> Sometimes menu does not open on new page

I've seen this off and on too. My guess is it's something to do with the way I'm overriding the (W) button not being very resilient. Now that I can use GB 8.2 I'm going to be switching how the overriding is done to using an official hook so 🤞 

> White on grey contrast

I'd say that the design isn't final, but this is unnecessarily bad 😅 I'll fix this

> Should new pages have a "New Page" entry?

🤔 I think based on p2s I've seen I'd expect it to have an entry. It'll start off as _Untitled_ and will need to update when the user sets the page title. But I think having a selected item will help.

The list is sorted by most recently edited, so I think it should always be at the top. That way when the user saves the page the selection will stay at the top of the list.

p-jackson commented:
@roo2 I've addressed those issues, as well as fixing posts that have html escapes in them. Haven't fixed the (W) not always working, I think that should be done in another PR.

p-jackson commented:
IE test failures are unrelated: Automattic/wp-calypso#43298

p-jackson commented on apps/full-site-editing/full-site-editing-plugin/wpcom-block-editor-nav-sidebar/src/style.scss:
```diff
@@ -99,21 +99,75 @@ $transition-period: 200ms;
+
+		.wpcom-block-editor-nav-sidebar__label {
+			background: $dark-gray-500;
+	}
```
Oh ha! I'm so lazy and rely on the tooling that I don't even notice when it doesn't work!

p-jackson commented:
Thanks @andrewserong yeah I just noticed the scrollbar thing too 😅 we're getting there but there's gonna be a lot of loose ends!
